### ISSUE TITLE
GeometrySystemGeneratorFast issues with use_sterics

### ISSUE THREAD

**Issue by jchodera** [choderalab/perses#308]
I enabled `use_sterics` and ran into this issue with `perses/tests/benchmark.py`:
```
[chodera@mskcc-ln1 perses.choderalab]$ cat ~/perses.o7833511
Warning: no access to tty (Bad file descriptor).
Thus no job control in this shell.
gpu-3-9-gpu0
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2016 NVIDIA Corporation
Built on Sun_Sep__4_22:14:01_CDT_2016
Cuda compilation tools, release 8.0, V8.0.44

Now testing butane null transformations
Running 0 two-stage ExpandedEnsemble steps for 200 iterations
Traceback (most recent call last):
  File "perses/tests/benchmark.py", line 192, in <module>
    benchmark_ncmc_work_during_protocol()
  File "perses/tests/benchmark.py", line 180, in benchmark_ncmc_work_during_protocol
    testsystem.exen_samplers[ENV].run(niterations=niterations)
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/samplers/samplers.py", line 1264, in run
    self.update()
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/samplers/samplers.py", line 1240, in update
    self.update_state()
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/samplers/samplers.py", line 1190, in update_state
    logp_accept, ncmc_new_positions = self._ncmc_geometry_ncmc(topology_proposal, positions, old_log_weight, new_log_weight)
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/samplers/samplers.py", line 1126, in _ncmc_geometry_ncmc
    geometry_new_positions, logP_forward = self._geometry_forward(topology_proposal, geometry_old_positions)
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/samplers/samplers.py", line 890, in _geometry_forward
    new_positions, geometry_logp_propose = self.geometry_engine.propose(topology_proposal, old_positions, self.sampler.thermodynamic_state.beta)
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 121, in propose
    logp_proposal, new_positions = self._logp_propose(top_proposal, current_positions, beta, direction='forward')
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 257, in _logp_propose
    growth_system_generator.set_growth_parameter_index(len(atom_proposal_order.keys())+1, context)
  File "/cbio/jclab/home/chodera/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 1823, in set_growth_parameter_index
    growth_force.updateParametersInContext(context)
  File "/cbio/jclab/home/chodera/miniconda3/lib/python3.5/site-packages/simtk/openmm/openmm.py", line 8666, in updateParametersInContext
    return _openmm.NonbondedForce_updateParametersInContext(self, context)
Exception: updateParametersInContext: The number of non-excluded exceptions has changed
```
I'll try to figure out what is going on and fix it.

**Comment by jchodera** [comment]
In fixing this, I realized there are currently *no* tests at all for `GeometrySystemGenerator` or `GeometrySystemGeneratorFast`. I'll create a new issue to remind us to add these.

**Comment by jchodera** [comment]
This is now fixed in #307, but when running the `perses/tests/test_geometry.py` tests to see whether it broke anything, I got this error:
```
======================================================================
ERROR: perses.tests.test_geometry_engine.test_propose_kinase_inhibitors
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/choderaj/miniconda3/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/tests/test_geometry_engine.py", line 1028, in test_propose_kinase_inhibitors
    run_proposals(proposals)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/tests/test_geometry_engine.py", line 1051, in run_proposals
    new_positions, logp = geometry_engine.propose(top_proposal, current_positions, beta)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 121, in propose
    logp_proposal, new_positions = self._logp_propose(top_proposal, current_positions, beta, direction='forward')
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 212, in _logp_propose
    growth_system_generator = GeometrySystemGeneratorFast(top_proposal.new_system, atom_proposal_order.keys(), growth_parameter_name, reference_topology=top_proposal.new_topology, use_sterics=self.use_sterics)
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 1766, in __init__
    self.set_growth_parameter_index(max(self._growth_indices))
  File "/Users/choderaj/github/choderalab/perses/perses.choderalab/perses/rjmc/geometry.py", line 1796, in set_growth_parameter_index
    if (growth_index < this_growth_index):
  File "/Users/choderaj/miniconda3/lib/python3.5/site-packages/parmed/topologyobjects.py", line 932, in __lt__
    return self.idx < other.idx
AttributeError: 'int' object has no attribute 'idx'
-------------------- >> begin captured logging << --------------------
```

**Comment by jchodera** [comment]
I'm beginning to think that having some tests disabled in travis is a Bad Idea (TM) and just leads to bitrot.

**Comment by jchodera** [comment]
Nope, that one was a real test error and totally my fault!

**Comment by pgrinaway** [comment]
>I'm beginning to think that having some tests disabled in travis is a Bad Idea (TM) and just leads to bitrot.

I agree, even if this one was a mistake, it's best to have all tests running. Travis can't really accommodate that though. I wouldn't mind trying to get a testing harness as described in #248 going once I fix #310

**Comment by jchodera** [comment]
I was thinking of taking a stab at the Jenkins setup mentioned in #248 today!

**Comment by jchodera** [comment]
This is fixed in #307

**Comment by pgrinaway** [comment]
>I was thinking of taking a stab at the Jenkins setup mentioned in #248 today!

Sounds good!
